components/camera/videosource/webcam.go

```diff
@@ -203,7 +205,29 @@ func NewWebcamSource(ctx context.Context, attrs *WebcamAttrs, logger golog.Logge
 	constraints := makeConstraints(attrs, debug, logger)
 
 	if attrs.Path != "" {
-		return tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		cam, err := tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+		if err != nil {
+			return cam, err
+		}
+
+		utils2.PanicCapturingGo(func() {
+			ticker := time.NewTicker(time.Second / 2)
+			defer ticker.Stop()
+			defer utils2.TryClose(ctx, cam)
```

Comment:
Is this double close  okay?

Reply:
Is this still an issue now that I'm using `cam.Close` directly? I'm not sure I understand what's double-closed here. If this is in reference to the close done by `getStatusVideoSource`, `getStatusVideoSource` now removed.